tests/test_gamecontroller.py

```diff
@@ -1,137 +1,70 @@
-        gc.master_game_state.board.game_almost_over = 0
-
-        self.assertIs(gc.game_over(0, gc.master_game_state), False)
-
-    def test_game_over_not_over_deck_empty(self):
-        mock_deck = create_autospec(Deck)
-        mock_deck.__len__.return_value = 0
-        mock_player = create_autospec(Player)
-        mock_game_state = create_autospec(GameState)
-        mock_board = create_autospec(Board)
-        gc = GameController([mock_player, mock_player])
-        gc.deck = mock_deck
-        gc.master_game_state = mock_game_state
-        gc.master_game_state.board = mock_board
-        gc.master_game_state.board.fuse_tokens = 1
-        gc.master_game_state.board.game_almost_over = 0
-
-        self.assertIs(gc.game_over(1, gc.master_game_state), False)
-
-
-    # TODO: Test play_game fails if player cannot make_move (may belong in make_move?)
-    def test_play_game_last_move_wins(self):
-        # Because if the final move causes a win we should return true but the game should end with correct points
-        pass
+@pytest.fixture(scope='module')
+def mock_player():
+    return create_autospec(Player)
+
+
+@pytest.fixture(scope="module")
+def mock_deck():
+    return util.create_mock_deck(r_draw_card='card')
+
+
+@pytest.fixture(scope='function')
+def gc(mock_player):
+    _gc = GameController([mock_player, mock_player])
+    _gc.master_game_state = util.create_mock_gamestate()
+    return _gc
+
+
+@pytest.mark.parametrize(argnames='player_count', argvalues=[1, 6])
+def test_constructor_fail_invalid_number_of_players(player_count, mock_player):
+    with pytest.raises(ValueError) as excinfo:
+        GameController(players=[mock_player for _ in range(player_count)])
+    assert str(excinfo.value) == 'There must be between 2 and 5 players to play Hanabi.'
+
+
+@pytest.mark.parametrize(argnames=['player_count', 'expected_hand_size'], argvalues=[[2, 5], [3, 5], [4, 4], [5, 4]])
+def test_deal_initial_hand_2p(player_count, expected_hand_size, mock_player, mock_deck):
+    gc = GameController([mock_player for _ in range(player_count)])
+    gc.deck = mock_deck
+    gc.deal_initial_hand()
+    assert len(gc.player_hands) == player_count
+    assert len(gc.player_hands[0]) == expected_hand_size
+
+
+# TODO: game_over: every escape case works, returns false when they have expected vals
+def test_game_over_no_fuse_tokens(gc):
+    gc.master_game_state.board = util.create_mock_board(fuse_tokens=0)
+    assert gc.game_over(0, gc.master_game_state)
+
+
+def test_game_over_all_stacks_completed(gc):
+    gc.master_game_state.board = util.create_mock_board(fuse_tokens=1,
+                                                        # Max score for a default game. Need to update when we add
+                                                        # wilds / configurable color counts.
+                                                        r_compute_score=25)
+    assert gc.game_over(0, gc.master_game_state)
```

Comment:
why drop assertIs ... True? Here and on line 45?

Reply:
Originally, the test used `TestCase.assertIs`. Without being contained in a subclass of `TestCase` the test doesn't have access to `assertIs`. I briefly looked for a `py.test` replacement but didn't find one.